thorlcr/master/thgraphmanager.cpp

```diff
@@ -1215,35 +1218,42 @@ void closeThorServerStatus()
  *  0 = unrecognised format, or wuid mismatch
  *  1 = success. new graph/wuid received.
  */
-static int recvNextGraph(unsigned timeoutMs, const char *wuid, StringBuffer &retWuid, StringBuffer &retGraphName)
+static int recvNextGraph(IJobQueue *thorQueue, unsigned timeoutMs, const char *wuid, StringBuffer &retWuid, StringBuffer &retGraphName)
 {
-    PROGLOG("Lingering time left: %.2f", ((float)timeoutMs)/1000);
-    CMessageBuffer msg;
-    if (!queryWorldCommunicator().recv(msg, NULL, MPTAG_THOR, nullptr, timeoutMs))
-        return -1;
     StringBuffer next;
-    msg.read(next);
+    CMessageBuffer msg;
+    if (thorQueue)
+    {
+        Owned<IJobQueueItem> item = thorQueue->dequeue(timeoutMs);
+        if (!item)
+            return -1;
+        next.set(item->queryWUID());
+    }
+    else
+    {
+        if (!queryWorldCommunicator().recv(msg, NULL, MPTAG_THOR, nullptr, timeoutMs))
+            return -1;
+        msg.read(next);
+    }
 
     // validate
     StringArray sArray;
     sArray.appendList(next, "/");
     if (2 == sArray.ordinality())
     {
-        if (wuid && !streq(sArray.item(0), wuid))
-            return 0;
-        msg.clear().append(true);
-        if (queryWorldCommunicator().reply(msg, 60*1000)) // should be quick!
+        if (!thorQueue)
         {
-            retWuid.set(sArray.item(0));
-            retGraphName.set(sArray.item(1));
-            return 1;
+            if (wuid && !streq(sArray.item(0), wuid))
+                return 0; // mismatch/ignore
+            msg.clear().append(true);
+            if (!queryWorldCommunicator().reply(msg, 60*1000)) // should be quick!
+                return -2; // failed to reply to client
         }
-        else
-            return -2;
     }
-    return 0;
+    retWuid.set(sArray.item(0));
```

Comment:
no longer returns 0 if `2 != sArray.ordinality()`.  Does it matter?

Reply:
change not intentional. It should return 0, meaning unrecognised format, ignore and read next.
I'll change.

Reply:
changed to return 0 (as before), meaning ignore unrecognised format